Modules/_winapi.c

```diff
@@ -772,23 +772,116 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject* sortenvironmentkey(PyObject *module, PyObject *item) {
+    Py_ssize_t n;
+    wchar_t* s = PyUnicode_AsWideCharString(item, &n);
+    if (s == NULL) {
+        return NULL;
+    }
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT, LCMAP_UPPERCASE, s);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    ""
+};
+
+static PyObject *
+normalize_environment(PyObject* environment) {
+    PyObject *result, *keys, *keyfunc, *sort, *args, *kwargs;
+
+    keys = PyMapping_Keys(environment);
+    if (keys == NULL) {
+        return NULL;
+    }
+
+    keyfunc = PyCFunction_New(&sortenvironmentkey_def, NULL);
+    sort = PyObject_GetAttrString(keys, "sort");
+    args = PyTuple_New(0);
+    kwargs = PyDict_New();
```

Comment:
```suggestion
```
We can pass kwargs as null

Reply:
Since we're using kwargs to pass the `key` parameter, so I kwargs should not be null, I reverted the commits